feature-libs/product-configurator/rulebased/components/tab-bar/configurator-tab-bar.component.ts

```diff
@@ -204,8 +214,57 @@ export class ConfiguratorTabBarComponent {
     }
   }
 
+  private isHandlingBackNavigation = false;
+  private routerEventsSub: Subscription;
+
   constructor(
     protected configRouterExtractorService: ConfiguratorRouterExtractorService,
-    protected configuratorCommonsService: ConfiguratorCommonsService
+    protected configuratorCommonsService: ConfiguratorCommonsService,
+    protected router: Router
   ) {}
+
+  ngOnInit(): void {
+    this.configuration$.pipe(take(1)).subscribe();
+
+    this.routerEventsSub = this.router.events.subscribe((event) => {
+      if (
+        event instanceof NavigationStart &&
+        event.navigationTrigger === 'popstate'
+      ) {
+        if (
+          !this.isHandlingBackNavigation &&
+          event.url.includes('/configure/') &&
+          this.router.url.includes('/configure-overview')
+        ) {
+          this.isHandlingBackNavigation = true;
```

Comment:
That doesn't look that nice, how do we know that this will be always reset? (Only from knowledge about the router, right?). Having this 'imperative' piece in an event handler is a smell (but TBH I don't know how to do it differently).